doc/source/migration_details.rst

```diff
@@ -5,36 +5,126 @@ Check out the code manually and upgrade your database by calling odoo-bin,
+********************************************
+
+Run the upgrade and check for errors. You will probably learn a lot about
+your data and have to do some manual clean up before and after the upgrade.
+Expect to repeat the process several times as you encounter errors, clean up
+your data, and try again. If necessary, ask for help or report bugs on
+Github.
+
+Add your migration scripts
+--------------------------
+
+There are plenty of examples in the code of existing migration scripts within
+OpenUpgrade for your current version and from the previous versions.
+
+You can also refer to the following documents:
+
+- `The Odoo Migration Manager
+  <https://doc.therp.nl/openupgrade/migrationmanager.html>`_
+- `OpenUpgrade API <https://doc.therp.nl/openupgrade/API.html>`_
+- the code of OpenUpgrade API
+  `openupgrade.py <https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py>`_
+- `Development FAQ <https://doc.therp.nl/openupgrade/devfaq.html>`_
+
+The Trial and error process for the development of your scripts
+---------------------------------------------------------------
+
+- [A] get the copy of your database in old version as `db-upgrade`, it is
+  easy to do through the database manager of your old odoo
+  `http://localhost:8012/web/database/manager`
+- [B] run `odoo -d db-upgrade -u all --stop-after-init`
+- [C] In case of error, fix the error (adding or editing migration
+  scripts), then rerun **without** the `--upgrade all`:
+  `odoo -d db-upgrade --stop-after-init`. This will continue the upgrade
+  process from where it left off, starting with the module that caused the
+  error. That way, the migration will not run the end-migration scripts of
+  the modules that were already upgraded before the error, so in case of a
+  live migration you always need to restore the database. But running this way
+  will help you save a lot of time in the upgrade process.
+
+Note that for the time being [C] is not available in all versions
+(see explanations here: https://github.com/OCA/OpenUpgrade/pull/2499).
+It is available for 12 and 13:
+https://github.com/OCA/OpenUpgrade/pulls?q=is%3Apr+%5BFIX%5D+reset+exception
+
+Whenever facing unexpected errors, you might want to restart from [A] as
+the step [C] might leave your database in an inconsistent state.
+
+Restart upgrade of the failed module instead of upgrading all systematically
+----------------------------------------------------------------------------
+
+*This can be seen as an alternative to the step [C] mentioned above.*
+
+Try `--update=all` first and then when it fails on a given module
+some_module, try to fix the scripts and do a ``--update=<some_module>``
+
+This enables fixing and testing the some_module scripts
+
+After it is fixed go back to ``--update=all`` because just doing
+``--update=<some_module>`` and letting dependent modules is likely to lead
+to partial and incomplete migration.
+
+``fsync = off`` in your postgresql.conf
```

Comment:
I consider this development advice, or at least `severely expert` mode as well because it is incredibly dangerous and I would very much hesitate to encourage people to do this, given that they might use the same database server for migration as for production. We already encounter a lot of cases where less experienced people get lost trying to work with openupgrade. Having them modify database settings like this could increase the collateral damage.

Reply:
Pedro's comment https://github.com/OCA/OpenUpgrade/pull/2606#discussion_r593771503 applies to this section.

Reply:
yeah, I was mentionning the fsync=off hack for experts only, I don't advise putting fsync=off inside the general OpenUpgrade documentation either. Otherwise some people would end up using the database wise that configuration later, get some energy cut and loose their data and cry after us.

Reply:
Hi @rvalyi. Sorry I misunderstood [your comment](https://github.com/OCA/OpenUpgrade/issues/2605#issuecomment-795319768) so.
I abstain on that point.
regards.